cmd/proxy.go

```diff
@@ -62,13 +62,16 @@ func (pc *proxyClient) serve(ctx context.Context) error {
 	ctx, cancel := context.WithCancel(ctx)
 	defer cancel()
 	exitCh := make(chan error)
-	defer close(exitCh)
 	for _, m := range pc.mnts {
 		go func(mnt *socketMount) {
 			err := pc.serveSocketMount(ctx, mnt)
 			if err != nil {
-				exitCh <- err
-				return
+				select {
+				// Best effort attempt to send error
```

Comment:
nit: I'd consider extending this comment a bit more to clarify that the only time this doesn't send "instantly" if an error has already been sent, and (I think) we only care about the first one.